server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectoryFactory.java

```diff
@@ -38,17 +40,25 @@ public RemoteSegmentStoreDirectoryFactory(Supplier<RepositoriesService> reposito
     @Override
     public Directory newDirectory(IndexSettings indexSettings, ShardPath path) throws IOException {
         String repositoryName = indexSettings.getRemoteStoreRepository();
+        String indexUUID = indexSettings.getIndex().getUUID();
+        String shardId = String.valueOf(path.getShardId().getId());
+        return newDirectory(repositoryName, indexUUID, shardId);
+    }
+
+    public Directory newDirectory(String repositoryName, String indexUUID, String shardId) throws IOException {
         try (Repository repository = repositoriesService.get().repository(repositoryName)) {
             assert repository instanceof BlobStoreRepository : "repository should be instance of BlobStoreRepository";
             BlobPath commonBlobPath = ((BlobStoreRepository) repository).basePath();
-            commonBlobPath = commonBlobPath.add(indexSettings.getIndex().getUUID())
-                .add(String.valueOf(path.getShardId().getId()))
+            commonBlobPath = commonBlobPath.add(indexUUID)
+                .add(shardId)
                 .add("segments");
 
             RemoteDirectory dataDirectory = createRemoteDirectory(repository, commonBlobPath, "data");
-            RemoteDirectory metadataDirectory = createRemoteDirectory(repository, commonBlobPath, "metadata");
+            RemoteDirectory metadataDirectory = createRemoteDirectory(repository, commonBlobPath,"metadata");
+            RemoteStoreMDShardLockManager mdLockManager = RemoteStoreMDLockManagerFactory.newLockManager(
```

Comment:
IMO, segment store directory should not be aware of the lock manager.

Reply:
as mentioned above, before acquiring shard lock, we need to make sure that the corresponding md file is present or not, that we are doing using remote segment store directory.